moveit_ros/moveit_servo/src/servo_parameters.cpp

```diff
@@ -77,175 +80,381 @@ ServoParameters::setParametersCallback(std::vector<rclcpp::Parameter> parameters
   {
-    RCLCPP_WARN(logger, "At least one of publish_joint_positions / "
+    RCLCPP_WARN(LOGGER, "At least one of publish_joint_positions / "
                         "publish_joint_velocities / "
                         "publish_joint_accelerations must be true. Check "
                         "yaml file.");
-    return nullptr;
+    return std::nullopt;
   }
-  if ((parameters->command_out_type == "std_msgs/Float64MultiArray") && parameters->publish_joint_positions &&
-      parameters->publish_joint_velocities)
+  if ((parameters.command_out_type == "std_msgs/Float64MultiArray") && parameters.publish_joint_positions &&
+      parameters.publish_joint_velocities)
   {
-    RCLCPP_WARN(logger, "When publishing a std_msgs/Float64MultiArray, "
+    RCLCPP_WARN(LOGGER, "When publishing a std_msgs/Float64MultiArray, "
                         "you must select positions OR velocities.");
-    return nullptr;
+    return std::nullopt;
   }
   // Collision checking
-  if (parameters->self_collision_proximity_threshold <= 0.)
+  if (parameters.self_collision_proximity_threshold <= 0.)
   {
-    RCLCPP_WARN(logger, "Parameter 'self_collision_proximity_threshold' should be "
+    RCLCPP_WARN(LOGGER, "Parameter 'self_collision_proximity_threshold' should be "
                         "greater than zero. Check yaml file.");
-    return nullptr;
+    return std::nullopt;
   }
-  if (parameters->scene_collision_proximity_threshold <= 0.)
+  if (parameters.scene_collision_proximity_threshold <= 0.)
   {
-    RCLCPP_WARN(logger, "Parameter 'scene_collision_proximity_threshold' should be "
+    RCLCPP_WARN(LOGGER, "Parameter 'scene_collision_proximity_threshold' should be "
                         "greater than zero. Check yaml file.");
-    return nullptr;
+    return std::nullopt;
   }
-  if (parameters->scene_collision_proximity_threshold < parameters->self_collision_proximity_threshold)
+  if (parameters.scene_collision_proximity_threshold < parameters.self_collision_proximity_threshold)
   {
-    RCLCPP_WARN(logger, "Parameter 'self_collision_proximity_threshold' should probably be less "
+    RCLCPP_WARN(LOGGER, "Parameter 'self_collision_proximity_threshold' should probably be less "
                         "than or equal to 'scene_collision_proximity_threshold'. Check yaml file.");
   }
-  if (parameters->collision_check_rate <= 0)
+  if (parameters.collision_check_rate <= 0)
   {
-    RCLCPP_WARN(logger, "Parameter 'collision_check_rate' should be "
+    RCLCPP_WARN(LOGGER, "Parameter 'collision_check_rate' should be "
                         "greater than zero. Check yaml file.");
-    return nullptr;
+    return std::nullopt;
   }
+  return parameters;
+}
 
-  // register parameter change callback
-  if (dynamic_parameters)
+ServoParameters::SharedConstPtr ServoParameters::makeServoParameters(const rclcpp::Node::SharedPtr& node,
+                                                                     const rclcpp::Logger logger,  // remove?
```

Comment:
TBD: Probably can remove this now that we don't need logger in the constructor. Removing this will mean we have to edit function calls in other files as well.